### PR TITLE
Update `accessor.py` to make `x.str.cat()` more flexible

### PR DESCRIPTION
**Pull request by samuelkampa**
- Removes condition on `x.str.cat()` that `others != None`
- Allows concatenation of strings from a single Series, mirroring [pandas functionality](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.str.cat.html))

Feel free to suggest changes that would make the proposed merge more performant.

### CODE REVIEW

File: dask/dataframe/accessor.py [hunk 1]
```diff
@@ -137,16 +137,23 @@ def split(self, pat=None, n=-1, expand=False):
     def cat(self, others=None, sep=None, na_rep=None):
         from .core import Series, Index
 
+        valid_types = (Series, Index, pd.Series, pd.Index)
+
         if others is None:
-            raise NotImplementedError("x.str.cat() with `others == None`")
+            if not isinstance(self, valid_types):
```
Comment (TomAugspurger) [inline dask/dataframe/accessor.py]: This shouldn't be needed.

File: dask/dataframe/accessor.py [hunk 2]
```diff
@@ -137,16 +137,23 @@ def split(self, pat=None, n=-1, expand=False):
...
+                raise TypeError("self must be Series/Index")
+            return self._series.map_partitions(
+            str_cat, meta=self._series._meta
+        ).compute().str.cat(sep=sep, na_rep=na_rep)
```
Comment (TomAugspurger) [inline dask/dataframe/accessor.py]: I don't think this is ideal. I don't think this should `compute` here. I'd find it suprising for `.str.cat()` to be immediate but `.str.cat(others=...)` to be delayed.

If we want to support this, I think that with `others=None` we'd return a Delayed string.

**Comment by mrocklin** [comment]
@samuelkampa checking in.  Thank you for your contribution here.  Do you have time to handle the suggestions that @TomAugspurger made above?

**Comment by samuelkampa** [comment]
Thanks for the reminder. I should be able to get to this by EOW.

**Comment by samuelkampa** [comment]
@TomAugspurger I'm not sure if I'm calling `dask.delayed` properly. Sorry for the messiness; I'm a relative newcomer to Python.

**Comment by mrocklin** [comment]
It looks like this has stalled.  I'm going to close this out.  @samuelkampa if you come back to this just let us know and we'll reopen it.

**Pull request closed without merging**